WordPress/gutenberg
# Data Views: Add action for pages to set site homepage

## What?

Adds an action to set the home page from the site editor.

Fixes https://github.com/WordPress/gutenberg/issues/63666

## Why?

Currently there is no way to change the home setting (Settings > Reading) from the site editor. This PR adds a way to do that.

## How?

Adds a dataview action for pages to set the page as the site homepage that can be triggered from the pages data view.

## Testing Instructions

1. Go to Editor > Pages.
2. Select the Actions menu
3. Select 'Set as homepage'
4. Ensure the selected page is set as the homepage

## Screenshots or screencast 

![image](https://github.com/user-attachments/assets/233d9ca6-2dea-4a3c-be72-4aecc5d9b5eb)

![image](https://github.com/user-attachments/assets/edb0f68f-a8e9-405b-b5bb-b42583f3765f)
